js/components/report/report.js

```diff
@@ -47,52 +62,53 @@ export default class Report extends PureComponent {
 
   renderStudentReports() {
     const { report, reportTree, sortedStudents } = this.props;
-    const nowShowing = report.get("nowShowing") === "student";
-    const className = nowShowing ? "report-content" : "report-content hidden";
-    const selectedStudentId = report.get("selectedStudentId");
+    if (report.get("nowShowing") !== "student") {
+      return null;
+    }
+    const selectedStudentIds = report.get("selectedStudentIds");
     let studentsToRender;
-    if (selectedStudentId) {
-      const selectedStudent = report.get("students").get(selectedStudentId);
-      if (selectedStudent) {
-        studentsToRender = [ selectedStudent ];
-      } else {
+    if (selectedStudentIds) {
+      studentsToRender = selectedStudentIds.map(sId => report.get("students").get(sId)).filter(s => !!s);
```

Comment:
I am afraid to ask what is happening here. 

Something very pathological about javascript boolean evaluation.  For what value is `!!a == true` but  `a == false`?   

Hold on while I go watch https://www.destroyallsoftware.com/talks/wat a few more times.

Reply:
Haha. 

>  For what value is !!a == true but a == false?
Never (I hope). 

I use `!!` to cast a value to `boolean`, e.g. values like `undefined` or `null` to `false` (even though they're "falsy" anyway, but TypeScript can complain about types). In this particular case `!!` is absolutely optional, I just want to filter out `undefined` values. `filter(s => s)` would work too. I just thought that casting to boolean will make it a bit more clear that this line/statement is not a mistake. Probably `s => s !== undefined` could work better for that.